add 'force_basic_auth' option

rule88 opened the pull request:
by default "force_basic_auth" option is set to False, with this adjustment it is up to the user if it wants to enforce basic authentication.

##### SUMMARY
Add support to the URL lookup plugin to support basic authentication (option: force_basic_auth)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugin --> lookup --> url

##### ADDITIONAL INFORMATION
This change is will help people needing to do an URL lookup with basic authentication enabled. By default it still is disabled, but with the force_basic_auth switch (boolean) it can be enabled.

rule88 committed:
add 'force_basic_auth' option

by default "force_basic_auth" option is set to False, with this adjustment it is up to the user if it wants to enforce basic authentication.

Akasurde commented on lib/ansible/plugins/lookup/url.py:
```diff
@@ -39,6 +39,10 @@
+  force_basic_auth:
+    description: Force basic authentication
+    type: boolean
+    version_added: "2.9"
```
```suggestion
    version_added: "2.10"
    default: False
```

Since 2.9 is Out.

Akasurde commented:
cc @sivel

rule88 committed:
Update lib/ansible/plugins/lookup/url.py

update version_added and set default to False

Co-Authored-By: Abhijeet Kasurde <akasurde@redhat.com>

rule88 commented:
@Akasurde already committed, but is it wise to have the default declared "again" in this plugin? It seems to already have a default in its module_utils:  https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py

samdoran commented:
This is ok, but adding options to the lookup one at a time will get out of hand. It would be better to go ahead and bring in all the options so the `uri` module has the same feature parity as the `url` lookup.

rule88 committed:
Adding multiple (optional) parameters

Parameters added:
force
timeout
http_agent
follow_redirects
use_gssapi
unix_socket
ca_path
unredirected_headers

rule88 commented:
@samdoran - added most "basic" parameters that are also to be found in the underlying used lib/ansible/module_utils/urls.py library:

- force
- timeout
- http_agent
- follow_redirects
- use_gssapi
- unix_socket
- ca_path
- unredirected_headers

 Left out the parameters where the datatypes don't seem to be ever passed by Ansible as Ansible afaik doesn't support passing those types. They also don't seem to add value to the "lookup" feature.

- data
- method ( out of this PR as this seems to not add value in a "lookup" module, 
- last_mod_time
- client_cert
- client_key
- cookies

rule88 committed:
Update url.py

rule88 committed:
Remove defaults where value is "None"

samdoran reviewed (changes requested):
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

rule88 committed:
add changelog fragment

rule88 commented:
@samdoran - changelog fragment added!

samdoran commented on changelogs/fragments/64892-add-parameters-to-url_lookup_plugin.yaml:
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - parameters - add parameters to url lookup plugin as they also exist in module_utils (urls.py).
```
```suggestion
  - url lookup plugin - add parameters to match what is available in ``module_utils/urls.py``
```

rule88 committed:
Update changelogs/fragments/64892-add-parameters-to-url_lookup_plugin.yaml

Co-Authored-By: Sam Doran <sdoran@redhat.com>

samdoran commented:
rebuild_merge

rule88 committed:
trigger rebuild
